### PR TITLE
Fail if the configuration file is missing

### PR DESCRIPTION
**Pull request by timgates42**
Addresses https://github.com/facelessuser/pyspelling/issues/80

### CODE REVIEW

**Comment by timgates42** [comment]
This caught me out when moving some files around and the CI step stopped warning me of spelling issues. Let me know if you don't agree or would like it changed.

**Comment by facelessuser** [comment]
I would ignore the appveyor failure which is unrelated to your changes. I'll fix that separately.

File: pyspelling/util/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """Utilities."""
-from __future__ import unicode_literals
+from __future__ import unicode_literals, print_function
```
Comment (facelessuser) [inline pyspelling/util/__init__.py]: We honestly don't need print_function as we only support python3. To be honest, importing of __future__ can be completely removed.

File: pyspelling/util/__init__.py [hunk 2]
```diff
@@ -170,14 +170,22 @@ class Loader(loader):
...
             if not file_name and name == '.spelling.yml':
                 warn_deprecated(
                     "Using '.spelling.yml' as the default is deprecated. Default config is now '.pyspelling.yml'"
                 )
             with codecs.open(name, 'r', encoding='utf-8') as f:
-                config = yaml_load(f.read())
-            break
-    return config
+                return yaml_load(f.read())
+    if not file_name:
```
Comment (facelessuser) [inline pyspelling/util/__init__.py]: We don't want to check file_name as we will default to the default file name if it isn't provided. A better check would probably be to initialize config to None and then check if it is None at the end.

File: pyspelling/util/__init__.py [hunk 3]
```diff
@@ -170,14 +170,22 @@ class Loader(loader):
...
+        file_name = '.pyspelling.yml'
+    print(
+        'Unable to load pyspelling configuration from %s, for more'
```
Comment (facelessuser) [inline pyspelling/util/__init__.py]: I would prefer if this raised an exception instead. You can use the same text for the exception though.

**Comment by facelessuser** [comment]
No worries. I have actually submitted a change for this which addresses this. I figured since I was already fixing Appveyor, I might as well just do this as well 🙂 .

**Comment by facelessuser** [comment]
Closed as this is fixed in #83

**Comment by timgates42** [comment]
Thanks for fixing it @facelessuser

**Pull request closed without merging**